### PR TITLE
fix: prevent macOS Sync Now timeouts on large histories

### PR DESCRIPTION
**Pull request by lamther**
## Summary

Prevent macOS Sync Now from timing out on large local histories by combining the bounded background scan with drain upload semantics, and preserve taps that arrive during a silent popover sync.

## Why

The native button currently sends only `{ "drain": true }`, which launches an exhaustive `sync --drain`. On a large Codex history (about 25 GB / 29k rollout files), the request reproduced the reported behavior:

- the child ran at high CPU for 120 seconds and exceeded 590 MB observed RSS;
- the embedded server returned `SYNC_TIMEOUT` / HTTP 500;
- queue and v2 cursor evidence did not advance;
- the UI stopped spinning without completing the requested refresh.

The existing bounded all-source path completed against the same data in about 4 seconds, with 218 MB observed RSS, and advanced `queue.jsonl`, the v2 manifest, and the current generation core.

## Changes

- Native Sync Now now requests `--auto --background --all-local-sources --publish-account --drain`.
- Explicit `--background` controls local scan breadth; `--drain` continues to control lock priority and full cloud backlog upload.
- Plain CLI `tracker sync --drain` remains an exhaustive local scan.
- A manual tap during silent popover sync is queued and visibly spins instead of being discarded by `syncInFlight`.

## Scope

- [x] CLI (`src/`)
- [ ] Dashboard (`dashboard/`)
- [x] macOS app (`TokenTrackerBar/`)
- [ ] Windows app (`TokenTrackerWin/`)
- [x] Docs / CI / config

## Verification

- `node --test --test-reporter=dot test/*.test.js`: 1731 passed in an isolated HOME.
- `xcodebuild ... -scheme TokenTrackerBarTests ... test`: 104 passed.
- `xcodebuild ... -scheme TokenTrackerBar ... build`: succeeded after XcodeGen + icon project patch.
- Targeted local API, lock, background-drain, and macOS contract tests: 52 passed.
- Live local API comparison confirmed the exhaustive timeout and bounded successful refresh described above.

## Checklist

- [x] `npm test` passes
- [x] No new user-facing strings
- [x] Commits follow conventional style
- [x] PR description explains why, not just what

---

<details open>
<summary><strong>Risk layer addendum</strong></summary>

### Risk layer triggers

- [ ] Public exposure / share links / unauthenticated access
- [ ] Auth / session / token handling
- [x] Cross-endpoint invariants or shared logic
- [x] External gateway / environment constraints

### Rules / invariants

- Plain `sync --drain` remains exhaustive for repair and initial backfill workflows.
- Only explicit `--auto --background` selects the bounded local scan.
- Drain still waits for lock priority, uploads up to the existing drain limit, and surfaces upload failure.
- Concurrent visible syncs coalesce; only a tap hidden behind a silent sync queues one follow-up.

### Boundary matrix

| Request | Local scan | Upload | Lock behavior |
|---|---|---|---|
| `sync --drain` | exhaustive | drain | priority wait |
| `sync --auto --background --drain` | bounded | drain | priority wait |
| `sync --auto --background` | bounded | existing background policy | existing behavior |
| tap during silent popover sync | queued once | manual semantics | starts after owner releases |

</details>

<details open>
<summary><strong>Codex review context</strong></summary>

- **Delta since last Codex review:** New branch from v0.83.5 / current main.
- **Intended behavior / invariants:** Keep user-triggered refresh fast without weakening CLI repair semantics or cloud drain readiness.
- **Edge cases covered:** large Codex histories, deep archives, lock contention, silent popover overlap, cloud-disabled and explicit-token local API paths.
- **Tests run:** full Node suite, full macOS XCTest suite, native App build, targeted integration tests, and live local API timing.
- **Known gaps / out of scope:** No change to the 120-second generic embedded child timeout; exhaustive CLI repair may still exceed it when called through a generic local API drain request.

### Most likely regression surface

- Local API argument forwarding for requests that combine `auto`, `background`, and `drain`.

### Verification method

- Behavioral Node tests assert process arguments and archive traversal.
- Swift policy tests cover all manual-request admission states.
- Full native compilation verifies ViewModel and APIClient integration.

### Uncovered scope

- No Windows UI behavior changed; Windows manual CLI sync remains exhaustive and is covered by the full suite.

</details>

## Summary by CodeRabbit

* **New Features**
  * Improved manual sync handling to start, queue, or combine requests based on current sync activity.
  * Sync Now now performs a lightweight local scan while also processing pending cloud uploads.
  * Background sync avoids deep archived-session scans while preserving upload processing.

* **Bug Fixes**
  * Corrected background and drain mode behavior when used together.

* **Tests**
  * Added coverage for sync request queuing, combined sync modes, and archived-session handling.

### CODE REVIEW

File: TokenTrackerBar/project.yml [hunk 1]
```diff
@@ -122,6 +122,7 @@ targets:
       - path: TokenTrackerBar/Models/LimitPace.swift
       - path: TokenTrackerBar/Models/WeeklyLimitResetDetector.swift
       - path: TokenTrackerBar/Models/BackgroundRefreshPolicy.swift
+      - path: TokenTrackerBar/Models/SyncRequestPolicy.swift
```
[the remark this answers is not in the document]
Comment (lamther) [reply TokenTrackerBar/project.yml:125]: Verified and intentionally not changed: TokenTrackerBar/TokenTrackerBar.xcodeproj/ is explicitly ignored by .gitignore. The repository contract requires running xcodegen generate plus scripts/patch-pbxproj-icon.rb, while CI regenerates the project in .github/workflows/ci.yml. Both commands were run locally, followed by a successful full TokenTrackerBar build and 104 passing macOS tests. Committing the generated project would conflict with the repository policy.

**Comment by mm7894215** [comment]
Thanks so much for this great fix!

Local end-to-end verification and performance testing confirm that combining the bounded background scan with drain upload semantics successfully prevents Sync Now timeouts and high CPU overhead on large histories.

Merged!
